neurovault/apps/statmaps/tasks.py

```diff
@@ -174,6 +174,65 @@ def generate_glassbrain_image(image_pk):
         img.thumbnail.save("glass_brain_%s.jpg" % img.pk, content_file)
         img.save()
 
+# SURFACE IMAGE GENERATION ###########################################################################
+
+@shared_task
+def generate_surfacce_image(image_pk):
+    from neurovault.apps.statmaps.models import Image
+    from scipy.io import loadmat
+    import numpy.matlib as matlib
+    from scipy.interpolate import interpn
+
+    img = Image.objects.get(pk=image_pk)
+    img_vol = nib.load(img.file.path)
+    data_vol = img_vol.get_data()
+    if data_vol.ndim > 3:
+        data_dim = data_vol.shape[3]  #number of time points
+    else:
+        data_dim = 1
+    this_path = os.path.abspath(os.path.dirname(__file__))
+
+    for hemi in ['lh', 'rh']:
+        if hemi == 'lh':
+            ras_coor = loadmat(os.path.abspath(os.path.join(this_path, "static", "anatomical",
+                                                            "lh.avgMapping_allSub_RF_ANTs_MNI2fs.mat")))['ras']
+        else:
+            ras_coor = loadmat(os.path.abspath(os.path.join(this_path, "static", "anatomical",
+                                                            "rh.avgMapping_allSub_RF_ANTs_MNI2fs.mat")))['ras']
+
+        vox2ras = img_vol.get_sform()
+        ras_centered = ras_coor - matlib.repmat(vox2ras[0:3, 3], ras_coor.shape[1], 1).T
+        vox_coor = numpy.dot(numpy.linalg.inv(vox2ras[0:3, 0:3]), ras_centered)  # convert to voxel coordinates
+
+        img_surf = nib.gifti.GiftiImage()
+        for i in range(data_dim):
+            if data_vol.ndim > 3:
+                data_curr = data_vol[:, :, :, i]
+                data_surf = interpn([range(data_vol.shape[0]), range(data_vol.shape[1]), range(data_vol.shape[2])],
+                                    data_curr, vox_coor.T, 'linear')
+                data_surf_gifti = nib.gifti.GiftiDataArray(data_surf, 'NIFTI_INTENT_TIME_SERIES', 'NIFTI_TYPE_FLOAT32',
+                                                           'ASCII')
```

Comment:
I noticed this line and field 'ASCII'. I believe most data is currently using 'GZipBase64Binary' instead, so you might consider changing it. It should be automatically parsed by nibabel either way.

Reply:
I've tried with other encodings, but they produce outputs that seem to be incompatible with Connectome Workbench (see https://github.com/nipy/nibabel/issues/642). Besides 'ASCII' surprisingly produces the smallest outputs.